paper-server/src/main/java/io/papermc/paper/world/biome/PaperBiomeSpecialEffects.java

```diff
@@ -0,0 +1,93 @@
+package io.papermc.paper.world.biome;
+
+import io.papermc.paper.util.MCUtil;
+import io.papermc.paper.world.biome.effects.AdditionSound;
+import io.papermc.paper.world.biome.effects.MoodSound;
+import io.papermc.paper.world.biome.effects.MusicEntry;
+import io.papermc.paper.world.biome.effects.PaperAdditionSound;
+import io.papermc.paper.world.biome.effects.PaperMoodSound;
+import io.papermc.paper.world.biome.effects.PaperMusicEntry;
+import net.minecraft.core.Holder;
+import net.minecraft.sounds.Music;
+import net.minecraft.util.random.WeightedList;
+import net.minecraft.world.level.biome.Biome;
+import org.bukkit.Color;
+import org.bukkit.Location;
+import org.bukkit.Sound;
+import org.bukkit.craftbukkit.CraftSound;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class PaperBiomeSpecialEffects implements BiomeSpecialEffects {
+    private final Holder<Biome> biome;
+
+    public PaperBiomeSpecialEffects(Holder<Biome> biome) {
+        this.biome = biome;
+    }
+
+    @Override
+    public Color fogColor() {
+        return Color.fromRGB(this.biome.value().getSpecialEffects().getFogColor());
```

Comment:
You should trim the alpha channel, even if it's not technically correct the codec allow just an int without color distinction so datapack with it can load fine.

Reply:
It's already trimmed by fromRGB

Reply:
Not really the method will throw an IllegalArgumentException in that case